### ISSUE TITLE
Feature request: preserve Ubam tags

### ISSUE THREAD

**Issue by matthdsm** [Illumina/DRAGMAP#8]
Hi, 

Would you be able to add a  flag that allows keeping the read tags when aligning an unaligned bam file, cfr the Bowtie2 `--preserve-tags` flag? Optionally with a list which tags to include/exclude.
This would allow us to skip merging the aligned bam and ubam post alignment, removing a bunch of overhead.

I'm looking for equivalency with dragmap + picard MergeBamAlignments as seen [here](https://github.com/broadinstitute/warp/blob/3e41edc997460dfd42dc53b4cbef4731c0751eda/tasks/broad/DragmapAlignment.wdl).

Thanks
M

**Comment by matthdsm** [comment]
ping @michaelgatzen as this might be relevant to your work over at WARP

**Comment by rizkg** [comment]
Hi,
Thank you for the suggestion, it indeed seems like a useful option when using uBAM files. We will look into it.

Guillaume

**Comment by tienpm7723** [comment]
Hi @matthdsm,

Im currently running warp for dragmap + picard MergeBamAlignments that you mentioned above but i got this error: 
Aligned record iterator (E00584:299:HGVWYCCX2:1:1101:10003:1122) is behind the unmapped reads (E00584:299:HGVWYCCX2:1:1101:10003:11224)

The problem is that dragen-os doesn't have --preserve-tags flag, isn't it? Thank you.

P/S: My input is uBam file created from 2 fastq file by using picard FastqToSam, I have tried using --preserve-map-align-order=1 but it doestn't work.

**Comment by matthdsm** [comment]
I don't think this issue has anything to do with this option, or `dragmap`, but stems from `picard`. I suggest you open an issue there.

M
